rpcs3/Emu/Cell/lv2/sys_storage.cpp

```diff
@@ -81,20 +135,207 @@ error_code sys_storage_get_device_info(u64 device, vm::ptr<StorageDeviceInfo> bu
+		buffer->sector_count = 0x4D955;
+		buffer->sector_size = 0x800;
+		buffer->one = 1;
+		buffer->flags[1] = 0;
+		buffer->flags[2] = 1;
+		buffer->flags[7] = 1;
+	}
+	else if (storage == USB_MASS_STORAGE_1(0))
+	{
+		if (dev_num > 0)
+		{
+			return not_an_error(-5);
+		}
+
+		std::string u = "unnamed";
+		memcpy(buffer->name, u.c_str(), u.size());
+		/*buffer->sector_count = 0x4D955;*/
+		buffer->sector_size = 0x200;
+		buffer->one = 1;
+		buffer->flags[1] = 0;
+		buffer->flags[2] = 1;
+		buffer->flags[7] = 1;
+	}
+	else if (storage == NAND_FLASH)
+	{
+		if (dev_num > 6)
+		{
+			return not_an_error(-5);
+		}
+
+		std::string u = "unnamed";
+		memcpy(buffer->name, u.c_str(), u.size());
+		buffer->sector_size = 0x200;
+		buffer->one = 1;
+		buffer->flags[1] = 1;
+		buffer->flags[2] = 1;
+		buffer->flags[7] = 1;
+
+		// see ata_hdd for explanation
+		switch (dev_num)
+		{
+		case 0: buffer->sector_count = 0x80000;
+			break;
+		case 1: buffer->sector_count = 0x75F8;
+			break;
+		case 2: buffer->sector_count = 0x63E00;
+			break;
+		case 3: buffer->sector_count = 0x8000;
+			break;
+		case 4: buffer->sector_count = 0x400;
+			break;
+		case 5: buffer->sector_count = 0x2000;
+			break;
+		case 6: buffer->sector_count = 0x200;
+			break;
+		}
+	}
+	else if (storage == NOR_FLASH)
+	{
+		if (dev_num > 3)
+		{
+			return not_an_error(-5);
+		}
+
+		std::string u = "unnamed";
+		memcpy(buffer->name, u.c_str(), u.size());
+		buffer->sector_size = 0x200;
+		buffer->one = 1;
+		buffer->flags[1] = 0;
+		buffer->flags[2] = 1;
+		buffer->flags[7] = 1;
+
+		// see ata_hdd for explanation
+		switch (dev_num)
+		{
+		case 0: buffer->sector_count = 0x8000;
+			break;
+		case 1: buffer->sector_count = 0x77F8;
+			break;
+		case 2: buffer->sector_count = 0x100; // offset, 0x20000
+			break;
+		case 3: buffer->sector_count = 0x400;
+			break;
+		}
+	}
+	else if (storage == NAND_UNK)
+	{
+		if (dev_num > 1)
+		{
+			return not_an_error(-5);
+		}
+
+		std::string u = "unnamed";
+		memcpy(buffer->name, u.c_str(), u.size());
+		buffer->sector_size = 0x800;
+		buffer->one = 1;
+		buffer->flags[1] = 0;
+		buffer->flags[2] = 1;
+		buffer->flags[7] = 1;
+
+		// see ata_hdd for explanation
+		switch (dev_num)
+		{
+		case 0: buffer->sector_count = 0x7FFFFFFF;
+			break;
+		}
+	}
+	else
+	{
+		sys_storage.error("sys_storage_get_device_info(device=0x%x, buffer=*0x%x)", device, buffer);
+	}
+
 	return CELL_OK;
 }
 
 error_code sys_storage_get_device_config(vm::ptr<u32> storages, vm::ptr<u32> devices)
 {
 	sys_storage.todo("sys_storage_get_device_config(storages=*0x%x, devices=*0x%x)", storages, devices);
 
+	*storages = 6;
+	*devices = 17;
+
 	return CELL_OK;
 }
 
 error_code sys_storage_report_devices(u32 storages, u32 start, u32 devices, vm::ptr<u64> device_ids)
 {
 	sys_storage.todo("sys_storage_report_devices(storages=0x%x, start=0x%x, devices=0x%x, device_ids=0x%x)", storages, start, devices, device_ids);
 
+	std::array<u64, 0x11> all_devs;
```

Comment:
This seems like it should a static constexpr initialized by an immediate lambda.